packages/rest/src/router/redirect-route.ts

```diff
@@ -19,15 +19,22 @@ export class RedirectRoute implements RouteEntry, ResolvedRoute {
 
   constructor(
     private readonly sourcePath: string,
-    private readonly targetLocation: string,
+    private readonly targetLocation: string | Function,
     private readonly statusCode: number = 303,
   ) {}
 
   async invokeHandler(
-    {response}: RequestContext,
+    {response,request}: RequestContext,
     args: OperationArgs,
   ): Promise<OperationRetval> {
-    response.redirect(this.statusCode, this.targetLocation);
+    let basePath = this._getBasePath(request) || '';
+    let targetLocation :string = typeof this.targetLocation === 'function'
+        ? this.targetLocation(
+          this._getProtocolForRequest(request),
+          this._getHost(request),
```

Comment:
We should not be running this complex logic for simple redirects. Here is how I am envisioning the implementation:

```ts
class {

  async invokeHandler(
    {response,request}: RequestContext,
    args: OperationArgs,
  ): Promise<OperationRetval> {
    if (typeof this.targetLocation === 'string') {
      response.redirect(this.statusCode, this._basePath + this.targetLocation);
      return;
    }
    const ctx: RedirectContext = {
      request,
      basePath: this._basePath,
      // build additional fields like protocol, request, etc.
    };
    const targetLocation = this.targetLocation(ctx);
    response.redirect(this.statusCode, targetLocation);
  }

}
```

See https://github.com/strongloop/loopback-next/issues/2022#issuecomment-468588124 for more details.